### PR TITLE
[Merged by Bors] - refactor(GroupTheory/Complement,SchurZassenhaus): Replace `Fintype.card` with `Nat.card`

### PR DESCRIPTION
**Pull request by tb65536**
We're in the home stretch!

This PR switches over `GroupTheory/Complement` and `GroupTheory/SchurZassenhaus` from `Fintype.card` to `Nat.card`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/SchurZassenhaus.lean [hunk 1]
```diff
@@ -232,20 +229,18 @@ private theorem step3 (K : Subgroup N) [(K.map N.subtype).Normal] : K = ⊥ ∨
   rwa [inj.eq_iff, inj.eq_iff] at key
 
 /-- Do not use this lemma: It is made obsolete by `exists_right_complement'_of_coprime` -/
-private theorem step4 : (Fintype.card N).minFac.Prime := by
-  rw [← Nat.card_eq_fintype_card]
+private theorem step4 : (Nat.card N).minFac.Prime := by
   exact Nat.minFac_prime (N.one_lt_card_iff_ne_bot.mpr (step0 h1 h3)).ne'
```
Comment (grunweg) [inline Mathlib/GroupTheory/SchurZassenhaus.lean:233]: Can you remove the "by exact" here?

**Review by grunweg — COMMENTED** [review]
Looks very reasonable (I'm not a group theory expert). Thanks for doing this!
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors d+

**Pull request closed without merging**